scripts/generate_secrets.py

```diff
@@ -2,6 +2,52 @@
 import yaml
 import sys
 
+import hashlib
+
+from yaml.representer import SafeRepresenter
+from cryptography.hazmat.backends.openssl.backend import backend
+from cryptography.hazmat.primitives.serialization import load_pem_private_key, Encoding, PublicFormat
+
+KEY_PURPOSE_AUTHENTICATION = 'authentication'
+KEY_PURPOSE_SUBMISSION = 'submission'
+
+# This script generates a yml file in the following format.
+# It is made up of two parts. string secrets (passwords) and keys (public and private)
+# keys:
+#   1234567890123456789012345678901234567890:
+#     purpose: submission
+#     type: public
+#     value: |
+#       -----BEGIN PUBLIC KEY-----
+#       #######################
+#       -----END PUBLIC KEY-----
+# secrets:
+#   PASSWORD: 'secret_value'
+
+
+class LiteralUnicode(str):
+    pass
+
+
+def change_style(style, representer):
+    """
+    This function is used to format the key value as a multi-line string maintaining the line breaks
+    """
+    def new_representer(dumper, data):
+        scalar = representer(dumper, data)
+        scalar.style = style
+        return scalar
+    return new_representer
+
+represent_literal_unicode = change_style('|', SafeRepresenter.represent_str)
+yaml.add_representer(LiteralUnicode, represent_literal_unicode)
+
+
+if len(sys.argv) < 2:
```

Comment:
Would be nice to introduce use of argparse under a if __name__ == "__main__" block. Argparse will handle your argument validation for you and it gives a distinct visual split between class/func defs and the code that is actually doing stuff.

Also means you can import the script for unit testing, if you want to.